go/cmd/dolt/dolt.go

```diff
@@ -420,8 +424,24 @@ func runMain() int {
 
 	_, usage := cli.HelpAndUsagePrinters(globalDocs)
 
-	apr, remainingArgs, err := globalArgParser.ParseGlobalArgs(args)
+	var fs filesys.Filesys
+	fs = filesys.LocalFS
+	dEnv := env.Load(ctx, env.GetCurrentUserHomeDir, fs, doltdb.LocalDirDoltDB, Version)
+	dEnv.IgnoreLockFile = ignoreLockFile
+
+	root, err := env.GetCurrentUserHomeDir()
+	if err != nil {
+		cli.PrintErrln(color.RedString("Failed to load the HOME directory: %v", err))
+		return 1
+	}
 
+	globalConfig, ok := dEnv.Config.GetConfig(env.GlobalConfig)
+	if !ok {
+		cli.PrintErrln(color.RedString("Failed to get global config"))
+		return 1
+	}
+
+	apr, remainingArgs, err := parseGlobalArgsAndProfile(globalConfig, args)
```

Comment:
This is going to be invoked for all commands - even commands which don't support global arguments yet. When you have a default, I believe this will make using some commands impossible because it will insert arguments into the command line when they shouldn't be there. You should add a test for this. Test that `dolt log` works when there is a default configuration